src/Core/src/Platform/Windows/SliderExtensions.cs

```diff
@@ -99,21 +103,52 @@ public static void UpdateThumbColor(this Slider platformSlider, ISlider slider)
 			"SliderThumbBackgroundDisabled",
 		};
 
-		internal static async Task UpdateThumbImageSourceAsync(this MauiSlider nativeSlider, ISlider slider, IImageSourceServiceProvider? provider)
+		internal static async Task UpdateThumbImageSourceAsync(this MauiSlider nativeSlider, ISlider slider, IImageSourceServiceProvider? provider, Size? defaultThumbSize)
 		{
 			var thumbImageSource = slider.ThumbImageSource;
 
 			if (thumbImageSource == null)
 			{
 				nativeSlider.ThumbImageSource = null;
+
+				var thumb = nativeSlider.GetFirstDescendant<Thumb>();
+
+				if (defaultThumbSize.HasValue && thumb is not null)
+				{
+					thumb.Height = defaultThumbSize.Value.Height;
+					thumb.Width = defaultThumbSize.Value.Width;
+				}
+
 				return;
 			}
 
 			if (provider != null && thumbImageSource != null)
 			{
 				var service = provider.GetRequiredImageSourceService(thumbImageSource);
 				var nativeThumbImageSource = await service.GetImageSourceAsync(thumbImageSource);
-
+				var nativeThumbImage = nativeThumbImageSource?.Value;
+
+				// BitmapImage is a special case that has an event when the image is loaded
+				// when this happens, we want to resize the thumb
+				if (nativeThumbImage is BitmapImage bitmapImage)
+				{
+					bitmapImage.ImageOpened += OnImageOpened;
+
+					void OnImageOpened(object sender, RoutedEventArgs e)
+					{
+						bitmapImage.ImageOpened -= OnImageOpened;
+
+						if (nativeSlider.GetFirstDescendant<Thumb>() is Thumb thumb)
```

Comment:
Same comment. I suspect that you have been using the is operator as a way to assign the output to a variable. It would be much cleaner in that case to change the method to have an out parameter and to a much nicer code such as:
```csharp
if (nativeSlider.TryGetFirstDescendant<Thumb> (var out thumb)) {
   // do the stuff
}
```

It is out of the scope of this PR.

Reply:
@jsuarezruiz if https://github.com/dotnet/maui/pull/13205 lands, the code can be made a lot nicer.

Reply:
Changed after merged https://github.com/dotnet/maui/pull/13205